README.md

```diff
@@ -1,101 +1,232 @@
+  ![completion-example](vscode-lean4/media/completion-example.png)
 
-## Functionality
-As of now, the extension supports the following features:
-- Diagnostics
-- Syntax highlighting with basic Lean 4 syntax rules
-- `\foo` unicode shortcuts
-- Semantic highlighting since [leanprover/lean4@5df753f3](https://github.com/leanprover/lean4/commit/5df753f3381dca714d0e81727d38feba765d6234) (2021-03-17)
-- An infoview since [leanprover/lean4@ec903f58](https://github.com/leanprover/lean4/commit/ec903f58d294cc7c57008a9cf754745fa7024f8c) (2021-02-04)
-- Go to definition since [leanprover/lean4@e627ad30](https://github.com/leanprover/lean4/commit/e627ad308d0d19d0407c0c7cb3ec82fb93fc12ea) (2021-01-19)
-- Type information & documentation on hover since [leanprover/lean4@d7c201a2](https://github.com/leanprover/lean4/commit/d7c201a2d4c4a0d0e76405e6d5325b63cb08ea11) (2021-01-15)
+- An Infoview displaying useful information about your current Lean program.
+- Go to definition (F12)
 - [Breadcrumbs](https://code.visualstudio.com/Docs/editor/editingevolved#_breadcrumbs)
-- Command to refresh file dependencies using Ctrl+Shift+X after recompilation
-- Command to restart the full Lean 4 server, accessible via Ctrl+Shift+P
-- Compatibility with the [Lean 3 VSCode extension](https://github.com/leanprover/vscode-lean)
+- `Lean4: Select Toolchain` command to select the toolchain you want to use instead of the default toolchain for your current workspace.  (**)
+
+    ![select-toolchain](vscode-lean4/media/select-toolchain.png)
+
+
+(*) Incremental updates do not yet work automatically across files, so after changing and rebuilding the dependency of a
+Lean 4 file, the language server needs to be manually informed that it should re-elaborate the full file, including the
+imports. This can be done using the `Lean 4: Refresh File Dependencies` command, which can be activated via <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>X</kbd>.
+
+(**) You can test the latest [Lean 4 master branch](https://github.com/leanprover/lean4) with this VS Code extension.  First you build Lean, then run the following commands:
+```
+cd <lean4-repo>
+cd ./build/release/stage1
+elan toolchain link master .
+```
+Now the `Lean4: Select Toolchain` will show `master` as one of the toolchains you can select.  The Lean Language Server will then be restarted automatically and will pick up your locally built version of Lean.  This is useful if you are adding features to Lean 4 that you want to test in the VS Code extension.
+
+## Lean editing features
+
+- Support for completing abbreviations starting with a backslash (\\).
+For example you type `\alpha` and the editor automatically replaces that with the nice Unicode character (α).
+- Support for completing the closing brace, like `()`, `{}` and `[]`
```

Comment:
If I'm not mistaken, this is just VSCode and not Lean specific.

Reply:
Ok, I updated the list with more that I think are lean specific because they are triggered by abbreviations.